subsys/usb/device/class/cdc_acm.c

```diff
@@ -1028,10 +1031,18 @@ static int cdc_acm_poll_in(const struct device *dev, unsigned char *c)
 static void cdc_acm_poll_out(const struct device *dev, unsigned char c)
 {
 	struct cdc_acm_dev_data_t * const dev_data = dev->data;
+	uint32_t wrote;
+	int lock;
 
 	dev_data->tx_ready = false;
 
-	while (!ring_buf_put(dev_data->tx_ringbuf, &c, 1)) {
+	while (true) {
+		lock = irq_lock();
+		wrote = ring_buf_put(dev_data->tx_ringbuf, &c, 1);
+		irq_unlock(lock);
+		if (wrote == 1) {
+			break;
+		}
```

Comment:
That is not a restriction imposed by the UART API.
`poll_out` is used by the console, so it would be difficult to make this restriction without reserving that UART instance for only console.